Assets/Scripts/Door.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Door : MonoBehaviour
+{
+    private bool open = false;
+    private bool moving = false;
+    
+    [Header("References")]
+    [SerializeField]
+    private Animator animator;
+    [SerializeField]
+    private ParticlesController explosionEffect;
+
+    public bool IsOpen(){ return open; }
+    public bool IsMoving(){ return moving; }
+    
+    //Plays opening animation;
+    public void Open(float speed = 1){
+        open = true;
+        moving = true;
+        animator.speed = speed;
+        animator.Play("DoorOpen");
```

Comment:
zamiast używać tekstu w funkcji, można dać string jako zmienna. W przypadku zmiany nazwy animacji/stanu czy innych podobnie wywoływanych funkcji, łatwiej jest to zmienić / debugować.